src/Build/Logging/BinaryLogger/BinaryLogger.cs

```diff
@@ -53,7 +53,9 @@ public sealed class BinaryLogger : ILogger
         //   - TargetSkippedEventArgs: added OriginallySucceeded, Condition, EvaluatedCondition
         // version 14:
         //   - TargetSkippedEventArgs: added SkipReason, OriginalBuildEventContext
-        internal const int FileFormatVersion = 14;
+        // version 15:
+        //   - Log our own environment variables by default. (This affects the message in the structured log viewer.)
```

Comment:
I don't understand this. There's no actual change, but you want a marker to change things in the viewer?

Reply:
This may end up reverted.

My original scheme was to use this to indicate we should change the string displayed when the user did not specify that they want all environment variables logged—it would say something like "we don't log env vars unless you set x," and it would now say "this list of env vars was truncated to just include y; set x to include them all." Sounds like KirillOsenkov thinks we can word it to make sense in either case. I owe him a reply.

Reply:
I was convinced, so I reverted this. To prevent version weirdness, I think we should make sure this makes it into 17.4.

Reply:
It's a tweak to alleviate a behavior difference in 17.4 so I think that's appropriate.